### PR TITLE
fix(download): Download chapter image fixes

### PR DESCRIPTION
**Pull request by SkillGG**
- Saving images in binary form, not utf-8 base64 so that gallery etc can see them
- Downloads work with new WebViewReader's image loading features
- Added loading icons when images are loading asynchronusely

### CODE REVIEW

File: src/database/queries/ChapterQueries.ts [hunk 1]
```diff
@@ -286,17 +286,37 @@ const downloadImages = async (
...
+            return '--';
+          })) +
           i +
           '.b64.png';
+        if (fileurl.charAt(0) === '-') {
+          return loadedCheerio.html();
+        }
         elem.replaceWith(
-          `<img src="${LoadingImageSrc}" class='loadIcon' file-path="${fileurl}">`,
+          `<img class='loadIcon' src="${LoadingImageSrc}" file-path="${fileurl}">`,
```
Comment (nyagami) [inline src/database/queries/ChapterQueries.ts:301]: we should use `load-icon` for class name instead

Comment (SkillGG) [reply src/database/queries/ChapterQueries.ts:301]: oh we can
I'll change that

File: src/screens/reader/components/WebViewReader.tsx [hunk 2]
```diff
@@ -124,7 +124,8 @@ const WebViewReader: React.FC<WebViewReaderProps> = props => {
                     const inject = datas.reduce((p, data) => {
                       return (
                         p +
-                        `document.querySelector("img[file-id='${data.id}']").setAttribute("src", "data:image/jpg;base64,${data.data}");`
+                        `document.querySelector("img[file-id='${data.id}']").setAttribute("src", "data:image/jpg;base64,${data.data}");
+                        document.querySelector("img[file-id='${data.id}']").classList.remove("loadIcon");`
```
Comment (nyagami) [inline src/screens/reader/components/WebViewReader.tsx:128]: so you forgor this

Comment (SkillGG) [reply src/screens/reader/components/WebViewReader.tsx:128]: Yeah.

**Review by nyagami — APPROVED** [review]

**Pull request merged**